02/app.js

```diff
@@ -1,7 +1,90 @@
 
 /* rozwiązanie z pętlą for */
-const x = 4;
 
+const getNumberFromUser = function () {
+    let x = prompt('Podaj liczbę od 1 do 9')
+    if (x === null || x.trim() === '') return
+    else if (isNaN(x)) {
+        alert('To nie jest liczba!');
 
+        return getNumberFromUser()
+    }
+    else if (x < 1 || x > 9) {
+        alert("Nieprawidłowy zakres. Wprowadź liczbę od 1 do 9")
 
-/* rozwiązanie z pętlą while  */
\ No newline at end of file
+        return getNumberFromUser()
+    }
+
+    return x
+}
+
+const number = getNumberFromUser()
```

Comment:
Mam wrażenie, że opcja z pętlą `do ... while` byłaby czytelniejsza i wygodniejsza niż wywoływanie samej siejbie - ale działa, to najważniejsze ;)